src/controllers/midi/midicontroller.cpp

```diff
@@ -245,27 +275,27 @@ void MidiController::receivedShortMessage(unsigned char status,
     if (isLearning()) {
         emit messageReceived(status, control, value);
 
-        auto it = m_temporaryInputMappings.constFind(mappingKey.key);
-        if (it != m_temporaryInputMappings.constEnd()) {
-            for (; it != m_temporaryInputMappings.constEnd() && it.key() == mappingKey.key; ++it) {
+        auto it = m_temporaryInputMappings.find(mappingKey.key);
+        if (it != m_temporaryInputMappings.end()) {
+            for (; it != m_temporaryInputMappings.end() && it.key() == mappingKey.key; ++it) {
                 processInputMapping(it.value(), status, control, value, timestamp);
             }
             return;
         }
     }
 
-    auto it = m_pMapping->getInputMappings().constFind(mappingKey.key);
-    for (; it != m_pMapping->getInputMappings().constEnd() && it.key() == mappingKey.key; ++it) {
+    auto it = m_pMapping->getInputMappings().find(mappingKey.key);
+    for (; it != m_pMapping->getInputMappings().end() && it.key() == mappingKey.key; ++it) {
```

Comment:
I'm also confused why this code even works in the first place. Could this be replaced by an `QMultiHash::equal_range`?

Reply:
About the constness, that's because [`processInputMapping` now modifies the `InputMapping`](https://github.com/mixxxdj/mixxx/pull/12781/files/b5bfd1677cfd4820dbfd966d4595a3aca972b1be#diff-ee87c49f05ff8aa897f732378f814f8e917cd4ec0b014b3021cda3c3e839dc7cR312)